content/go/CDN_services.go.md

```diff
@@ -0,0 +1,568 @@
+	if err != nil {
+		fmt.Printf("\n Unable to retrieve the cdn key name:\n - %s\n", err)
+		return
+	}
+	cdnPackageId := sl.Int(*cdnPackageKeyName[0].Id)
+
+	cdnItemPrices, err := productPackageService.Id(*cdnPackageId).GetItemPrices()
+	if err != nil {
+		fmt.Printf("\n Unable to get the cdn prices:\n - %s\n", err)
+		return
+	}
+	itemPriceId := sl.Int(*cdnItemPrices[0].Id)
+
+	prices := []datatypes.Product_Item_Price{
+		{Id: sl.Int(*itemPriceId)},
+	}
+	// Build a skeleton SoftLayer_Container_Product_Order_Network_ContentDelivery_Account object
+	// containing the order you wish to place.
+	// Container_Product_Order_Network_ContentDelivery_Account
+	templateObject := datatypes.Container_Product_Order{
+
+		PackageId: sl.Int(*cdnPackageId),
+		Prices:    prices,
+
+		}
+	jsonFormat, jsonErr := json.MarshalIndent(templateObject, "", "     ")
+	if jsonErr != nil {
+		fmt.Println(jsonErr)
+		return
+	}
+	fmt.Println(string(jsonFormat))
+	// Use verifyOrder() method to check for errors. Replace this with placeOrder() when
+	// you are ready to order.
+	receipt, err := service.PlaceOrder(&templateObject,sl.Bool(false))
+	if err != nil {
+		fmt.Printf("\n Unable to place order:\n - %s\n", err)
+		return
+	}
+
+	// Following helps to print the result in json format.
+	jsonFormat1, jsonErr1 := json.MarshalIndent(receipt, "", "     ")
+	if jsonErr1 != nil {
+		fmt.Println(jsonErr1)
+		return
+	}
+	fmt.Println(string(jsonFormat1))
+}
+```
+
+Note: Only create when no exist an account created or disabled
+
+### ListVendors 
+```
+package main
+import (
+"encoding/json"
+"fmt"
+"github.com/softlayer/softlayer-go/services"
+"github.com/softlayer/softlayer-go/session"
+)
+func main() {
+	// SoftLayer API username and key
+	username := "set-me"
+	apikey := "set-me"
+
+	// Create SoftLayer API session
+	sess := session.New(username, apikey)
+
+	service := services.GetNetworkCdnMarketplaceVendorService(sess)
+
+	receipt, err := service.ListVendors()
+	if err != nil {
+		fmt.Printf("\n Unable to place order:\n - %s\n", err)
+		return
+	}
+
+	// Following helps to print the result in json format.
+	jsonFormat1, jsonErr1 := json.MarshalIndent(receipt, "", "     ")
+	if jsonErr1 != nil {
+		fmt.Println(jsonErr1)
+		return
+	}
+	fmt.Println(string(jsonFormat1))
+}
+```
+
+### VerifyCdnAccountExists
+
+```
+package main
+
+import (
+"encoding/json"
+"fmt"
+"github.com/softlayer/softlayer-go/services"
+"github.com/softlayer/softlayer-go/session"
+"github.com/softlayer/softlayer-go/sl"
+)
+
+func main() {
+	// SoftLayer API username and key
+	username := "set - me"
+	apikey := "set - me"
+
+	// Create SoftLayer API session
+	sess := session.New(username, apikey)
+
+	service := services.GetNetworkCdnMarketplaceAccountService(sess)
+
+	vendor:=sl.String("akamai")
+
+	receipt, err := service.VerifyCdnAccountExists(vendor)
+	if err != nil {
+		fmt.Printf("\n Unable to place order:\n - %s\n", err)
```

Comment:
Make this error more generic, since this isn't placeing an order.

fmt.Printf("\nAPI Error:\n - %s\n", err)